test/clj/rems/api/test_applications.clj

```diff
@@ -1764,15 +1781,238 @@
                 "handler-approve.txt"
                 "handler-close.txt"
                 "handler-close (1).txt"]
-               (mapv :attachment/filename (:application/attachments (get-application-for-user application-id applicant-id))))))
+               (get-attachments application-id applicant-id))))
       (testing "handler"
         (is (= ["handler-public-remark.txt"
                 "reviewer-review.txt"
                 "handler-private-remark.txt"
                 "handler-approve.txt"
                 "handler-close.txt"
                 "handler-close (1).txt"]
-               (mapv :attachment/filename (:application/attachments (get-application-for-user application-id handler-id)))))))))
+               (get-attachments application-id handler-id)))))))
+
+(deftest test-application-redact-attachments
+  (let [api-key "42"
+        applicant-id "alice"
+        handler-id "handler"
+        reviewer-id "developer"
+        decider-id "carl"
+        file #(assoc filecontent :filename %)
+        workflow-id (test-helpers/create-workflow! {:type :workflow/decider
+                                                    :handlers [handler-id]})
+        cat-item-id (test-helpers/create-catalogue-item! {:workflow-id workflow-id})
+        application-id (test-helpers/create-application! {:catalogue-item-ids [cat-item-id]
+                                                          :actor applicant-id})
+        add-attachment #(-> (request :post (str "/api/applications/add-attachment?application-id=" application-id))
+                            (authenticate api-key %1)
+                            (assoc :multipart-params {"file" %2})
+                            handler
+                            read-ok-body
+                            :id)]
+    (testing "submit"
+      (is (= {:success true} (send-command applicant-id
+                                           {:type :application.command/submit
+                                            :application-id application-id}))))
+    (testing "unrelated user can't upload attachment"
+      (is (response-is-forbidden? (-> (request :post (str "/api/applications/add-attachment?application-id=" application-id))
+                                      (authenticate api-key reviewer-id)
+                                      (assoc :multipart-params {"file" filecontent})
+                                      handler))))
+    (testing "invite reviewer"
+      (is (= {:success true} (send-command handler-id
+                                           {:type :application.command/request-review
+                                            :application-id application-id
+                                            :reviewers [reviewer-id]
+                                            :comment "please"}))))
+    (testing "invite decider"
+      (is (= {:success true} (send-command handler-id
+                                           {:type :application.command/request-review
```

Comment:
Was this supposed to be `request-decision`?

Reply:
yeah, copy-paste error here. good catch :)